src/rust/src/backend/dh.rs

```diff
@@ -247,6 +247,17 @@ impl DHPrivateKey {
     fn __copy__(slf: pyo3::PyRef<'_, Self>) -> pyo3::PyRef<'_, Self> {
         slf
     }
+
+    fn __deepcopy__<'p>(
+        slf: pyo3::PyRef<'p, Self>,
+        _memo: &pyo3::Bound<'p, pyo3::types::PyDict>,
+    ) -> CryptographyResult<Self> {
+        let new_key = Self {
```

Comment:
You shouldn't need to do this, because keys are immutable a version of `return self` (like `__copy__`) should be enough.